scene/gui/line_edit.cpp

```diff
@@ -695,11 +701,11 @@ void LineEdit::_notification(int p_what) {
 		} break;
 		case NOTIFICATION_WM_FOCUS_OUT: {
 			window_has_focus = false;
-			draw_caret = false;
+			draw_caret = caret_force_displayed;
```

Comment:
I'm not sure about this one. If I'm correct, multi windows will be supported in the 4.0 version, and I'm not sure when this notification will be emitted. Anyway, it could be interesting to keep drawing the caret if the LineEdit can be controlled from an other window. 
About the if statement, you are right. But I didn't understand what you meant about the "other statements".

Reply:
`NOTIFICATION_WM_FOCUS_OUT` is emitted when the window containing the control loses focus, as such switching to another game window will also emit the notification. 

By other statements I mean where the bool `window_has_focus` is used i.e line 702 need to have the logic swapped so it becomes something like `(has_focus()...) || (!window_has_focus && !caret_force_display)` there are a couple more place that will also need this treatment.

Reply:
Ok, got it.